pkg/resource/deploy/import.go

```diff
@@ -233,9 +235,13 @@ func (i *importer) registerProviders(ctx context.Context) (map[resource.URN]stri
 			return nil, false, errors.New("incorrect package type specified")
 		}
 
+		providerName := imp.Type.Package()
 		parameterization := providers.ToProviderParameterization(imp.Parameterization)
+		if imp.BaseProviderName != "" {
+			providerName = tokens.Package(imp.BaseProviderName)
+		}
 		req := providers.NewProviderRequest(
-			imp.Type.Package(), imp.Version, imp.PluginDownloadURL, imp.PluginChecksums, parameterization)
+			providerName, imp.Version, imp.PluginDownloadURL, imp.PluginChecksums, parameterization)
 		typ, name := providers.MakeProviderType(req.Package()), req.DefaultName()
```

Comment:
Oh curses. Too many moving parts, this was wrong from https://github.com/pulumi/pulumi/pull/18038. Lemme pick this thread up, its horrible and needs to rejig some of 18038.